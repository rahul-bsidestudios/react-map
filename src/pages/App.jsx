import React, { useState } from 'react';
import Filters from '../components/filters';
import Loader from '../components/loader';
import Map from '../components/map';
import './app.css';

const convertPath = path => {
  const arr = [];
  path.map(item => {
    return arr.push([parseFloat(item[1]), parseFloat(item[0])]);
  });
  return arr;
}

const App = (props) => {
  const [loading, setLoading] = useState(false);
  const [path, setPath] = useState(null);
  const [origin, setOrigin] = useState('');
  const [destination, setDestination] = useState('');

  const createPath = (_path, _origin, _destination) => {
    setPath(convertPath(_path));
    setOrigin(_origin);
    setDestination(_destination);
  }

  const clear = () => {
    setPath(null);
    setLoading(false);
  }

  return (
    <div className="container-fluid">
      <div className="row">
        <div className="col-sm-4">
          <Filters
            setLoading={setLoading.bind(this)}
            clear={clear.bind(this)}
            createPath={createPath.bind(this)}
          />
        </div>
        <div className="col-sm-8">
          {loading && <Loader />}
          <Map path={path} origin={origin} destination={destination} />
        </div>
      </div>
    </div>
  );
}

export default App;
