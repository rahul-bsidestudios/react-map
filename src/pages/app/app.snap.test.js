import React from 'react';
import renderer from 'react-test-renderer';
import App from './index';

jest.mock('mapbox-gl/dist/mapbox-gl', () => ({
  Map: function () {
    this.on = jest.fn();
  }
}));

test('renders App component', () => {
  const tree = renderer.create(<App />).toJSON();
  expect(tree).toMatchSnapshot();
});