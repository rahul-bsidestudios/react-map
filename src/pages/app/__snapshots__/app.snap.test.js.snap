// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders App component 1`] = `
<div
  className="container-fluid"
>
  <div
    className="loading"
  >
    Loadingâ€¦
  </div>
  <div
    className="row"
  >
    <div
      className="col-sm-4"
    >
      <div
        className="filters"
      >
        <div
          className="form-group"
        >
          <label>
            Starting location
          </label>
          <div>
            <input
              autoComplete="off"
              className="form-control search react-mapbox-ac-input"
              onChange={[Function]}
              onKeyDown={[Function]}
              placeholder="Start Location"
              type="text"
              value=""
            />
            
            <span>
              <div
                className="react-mapbox-ac-menu"
                onClick={[Function]}
                style={
                  Object {
                    "display": "none",
                  }
                }
              >
                
              </div>
            </span>
          </div>
        </div>
        <div
          className="form-group last-control"
        >
          <label>
            Drop-off location
          </label>
          <div>
            <input
              autoComplete="off"
              className="form-control search react-mapbox-ac-input"
              onChange={[Function]}
              onKeyDown={[Function]}
              placeholder="Drop-off Location"
              type="text"
              value=""
            />
            
            <span>
              <div
                className="react-mapbox-ac-menu"
                onClick={[Function]}
                style={
                  Object {
                    "display": "none",
                  }
                }
              >
                
              </div>
            </span>
          </div>
        </div>
        
        
        <div
          className="buttons"
        >
          <button
            className="btn btn-primary"
            disabled={true}
            onClick={[Function]}
            type="button"
          >
            Loading
          </button>
          <button
            className="btn btn-secondary"
            disabled={true}
            onClick={[Function]}
            type="button"
          >
            Reset
          </button>
        </div>
      </div>
    </div>
    <div
      className="col-sm-8"
    >
      <div
        style={
          Object {
            "height": "100vh",
            "width": "100%",
          }
        }
      />
    </div>
  </div>
</div>
`;
